_source/_posts/2020-12-11-build-and-secure-an-api-in-python-with-fastapi.md

```diff
@@ -0,0 +1,362 @@
+---
+layout: blog_post
+title: "Build and Secure an API in Python with FastAPI"
+author: karl-hughes
+by: contractor
+communities: [python]
+description: "How to quickly build a REST API with FastAPI and secure it with Okta"
+tags: [python, fastapi, authentication]
+tweets:
+- "Learn how to quickly build a REST API in Python with FastAPI"
+- "Tired of the overhead it takes to build a REST API? Use FastAPI!"
+- "Quickly build out a database backed REST API with documentation using FastAPI"
+image: blog/build-and-secure-an-api-in-python-with-fastapi/J9aTg6J.png
+type: conversion
+---
+As [Python grows in popularity](https://www.zdnet.com/article/programming-language-popularity-python-overtakes-java-as-rust-reaches-top-20/), the variety of high-quality frameworks available to developers has blossomed. In addition to steadfast options like [Django](https://www.djangoproject.com/) and [Flask](https://flask.palletsprojects.com/en/1.1.x/), there are many new options including [FastAPI](https://fastapi.tiangolo.com/).
+
+First released [in late 2018](https://github.com/tiangolo/fastapi/releases?after=0.2.0), FastAPI differentiates itself from other Python frameworks by offering a modern, fast, and succinct developer experience for building reliable [REST APIs](/blog/2019/09/04/securing-rest-apis). While one of the newer open-source Python frameworks available, FastAPI has quickly gained a following with over [22,000 stars on GitHub](https://github.com/tiangolo/fastapi) and an active [community of maintainers](https://fastapi.tiangolo.com/fastapi-people/) working on the project.
+
+In this post, you'll learn more about FastAPI, and why it might be the right choice for your next project. You'll see how to set up a new FastAPI project and use [Okta](https://www.okta.com/) to secure the API.
```

Comment:
In previous times, Lindsay used to recommend we have no links in the opening paragraphs so people don't leave the post before they read it. It's probably too late to do it in this post, but I'd try to keep it in mind for future posts.

Reply:
Good to know. Thanks.